### PR TITLE
fix(retry): fix the retry policy do not take affect

### PR DESCRIPTION
**Pull request by whalecold**
#### What this PR does / why we need it (en: English/zh: Chinese):

The logic of retry is before the router, the value of key RouterClusterKey can't be found, add a new router object to the retry policy manager, use  the router to acquire the RouterClusterKey.
en: 
zh:

### CODE REVIEW

**Comment by ppzqh** [comment]
We can extent the retryer to use router logic and do the retry based on retry policy from RDS.

File: go.mod [hunk 1]
```diff
@@ -3,14 +3,16 @@ module github.com/kitex-contrib/xds
 go 1.17
 
 require (
-	github.com/bytedance/gopkg v0.0.0-20240514070511-01b2cbcf35e1
+	github.com/bytedance/gopkg v0.1.0
```
Comment (GuangmingLuo) [inline go.mod:6]: 用最新的 0.1.1 吧

File: go.mod [hunk 2]
```diff
@@ -3,14 +3,16 @@ module github.com/kitex-contrib/xds
 go 1.17
```
Comment (GuangmingLuo) [inline go.mod:3]: 这里可以用 go 1.18，和 kitex 保持一致

File: go.mod (hunk 1 above)
Comment (whalecold) [reply go.mod]: Done

File: go.mod (hunk 2 above)
Comment (whalecold) [reply go.mod]: Done.

**Comment by whalecold** [comment]
> We can extent the retryer to use router logic and do the retry based on retry policy from RDS.

Fixed, I will test in actual environment later

**Comment by ppzqh** [comment]
> > We can extent the retryer to use router logic and do the retry based on retry policy from RDS.
> 
> Fixed, I will test in actual environment later

ok👍

File: xdssuite/retry.go [hunk 3]
```diff
@@ -83,42 +90,80 @@ func updateRetryPolicy(rc *retrySuit, res map[string]xdsresource.Resource) {
...
+		updateRetryPolicy(rs, res)
 	})
-	return client.WithRetryContainer(retry.Container)
+	return client.WithRetryContainer(rs.Container)
 }
 
 // keep consistent when initialising the circuit breaker suit and updating
 // the retry policy.
-func genRetryServiceKey(ctx context.Context, ri rpcinfo.RPCInfo) string {
+func (rs *retrySuit) genRetryServiceKey(ctx context.Context, ri rpcinfo.RPCInfo) string {
```
Comment (ppzqh) [inline xdssuite/retry.go:137]: suite?

File: xdssuite/retry.go [hunk 4]
```diff
@@ -83,42 +90,80 @@ func updateRetryPolicy(rc *retrySuit, res map[string]xdsresource.Resource) {
...
 	m := xdsResourceManager.getManager()
 	if m == nil {
 		panic("xds resource manager has not been initialized")
 	}
 
-	retry := &retrySuit{
-		Container: retry.NewRetryContainer(retry.WithCustomizeKeyFunc(genRetryServiceKey)),
+	rs := &retrySuit{
+		router:      NewXDSRouter(opts...),
+		matchMethod: opt.matchRetryMethod,
```
Comment (ppzqh) [inline xdssuite/retry.go:125]: 只要配了这个，就一定要匹配method吗

File: xdssuite/option.go [hunk 5]
```diff
@@ -53,6 +55,33 @@ func NewOptions(opts []Option) *Options {
...
+func WithServicePort(port uint32) Option {
+	return Option{
+		F: func(o *Options) {
+			o.servicePort = port
+		},
+	}
+}
+
+// WithMatchRetryMethod configures the flag of matchRetryMethod
+func WithMatchRetryMethod(match bool) Option {
```
Comment (ppzqh) [inline xdssuite/option.go:68]: methodMethod?

File: xdssuite/limiter.go [hunk 6]
```diff
@@ -52,13 +52,13 @@ func getLimiterPolicy(up map[string]xdsresource.Resource) map[uint32]uint32 {
 	if lds == nil {
 		return nil
 	}
-	maxTokens := make(map[uint32]uint32)
+	tpfs := make(map[uint32]uint32)
```
Comment (ppzqh) [inline xdssuite/limiter.go:55]: limiter这个要不单独pr改一下？

Comment (whalecold) [reply xdssuite/limiter.go:55]: ok

File: xdssuite/retry.go (hunk 4 above)
Comment (whalecold) [reply xdssuite/retry.go:125]: 是的，感觉非幂等的接口使用重试策略会有问题，希望是客户只配置幂等的 method，然后也不要加兜底策略。

File: xdssuite/option.go (hunk 5 above)
Comment (whalecold) [reply xdssuite/option.go:68]: 是不是加个 retry 是不是更好，表示生效于哪种策略

File: xdssuite/option.go [hunk 7]
```diff
@@ -71,15 +100,16 @@ func (c *clientSuite) Options() []client.Option {
 }
 
 // NewClientSuite client suite for xds handler
-func NewClientSuite(opts ...Option) *clientSuite {
+func NewClientSuite(opts ...Option) client.Option {
```
Comment (ppzqh) [inline xdssuite/option.go:103]: 这个是 breaking change 了

**Review by ppzqh — APPROVED** [review]

**Pull request merged**